### ISSUE TITLE
matchplay false postivies cannot be disabled with # noqa

### ISSUE THREAD

**Issue by cornydesouza** [ansible-community/ansible-lint#1839]
##### Summary
False positives reported by matchplay cannot be disabled with # noqa rule-name

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
```console
ansible --version
ansible [core 2.11.5] 
  config file = None
  configured module search path = ['/home/qc/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  ansible collection location = /home/qc/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 3.0.1
  libyaml = True
ansible-lint --version
ansible-lint 5.1.3 using ansible 2.11.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT
```
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal
```

##### STEPS TO REPRODUCE

```console
ansible-lint -p -t var-naming
```

playbook.yml
```yml
---
- name: var-naming issue
  vars:
    test-var: testValue # noqa var-naming
  hosts:
    - localhost
  roles:
    - test
```

##### Desired Behaviour
The invalid variable name should be ignored due to the # noqa var-naming. Note var-naming is just an example rule that is used to illustrate the issues. As far as I can tell this is the case for all rules that rule matchplay.

##### Actual Behaviour

```console
WARNING  Listing 1 violation(s) that are fatal
playbook.yml:4: var-naming Play defines variable 'test-var' within 'vars' section that violates variable naming standards
You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental

Finished with 0 failure(s), 1 warning(s) on 1 files.

```
[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
